rasterio/plot.py

```diff
@@ -33,38 +33,69 @@
 logger = logging.getLogger(__name__)
 
 
-def show(source, cmap='gray', with_bounds=True):
+def show(source, cmap='gray', with_bounds=True, ax=None, title=None, **kwargs):
     """Display a raster or raster band using matplotlib.
 
     Parameters
     ----------
-    source : array-like or (raster dataset, bidx)
+    source : array-like, or (raster dataset, bidx) tuple, or raster dataset, 
         If array-like, should be of format compatible with
         matplotlib.pyplot.imshow. If the tuple (raster dataset, bidx),
-        selects band `bidx` from raster.
+        selects band `bidx` from raster.  If raster dataset display the rgb image 
+        as defined in the colorinterp metadata, or default to first band. 
     cmap : str (opt)
         Specifies the colormap to use in plotting. See
         matplotlib.Colors.Colormap. Default is 'gray'.
     with_bounds : bool (opt)
         Whether to change the image extent to the spatial bounds of the image,
         rather than pixel coordinates. Only works when source is
-        (raster dataset, bidx).
+        (raster dataset, bidx) or raster dataset.
+    ax : matplotlib axes (opt)
```

Comment:
I had added a return of the ax which introduced this. Turns out you can't call plt.gca() after show.  It's fixed now.

I agree that more test are needed at this point.  I'll see what I can do to get them added.

This notebook was added to the examples folder.  It could probably use a bit more cleanup and description to be a fully fleshed out example.  I'll work on adding that as well.

@brendan-ward, I'm not sure it's a formalized convention but ax as shorthand for an axes is used extensively.  It's a bit confusing but in mpl the term 'axis' refers to the x or y axis and 'axes' refers to the subplot on a figure: http://i.stack.imgur.com/HZWkV.png.  You also see axes being used for a list of axes so that you have axes[i].  But I think what I've implemented is fairly conventional.